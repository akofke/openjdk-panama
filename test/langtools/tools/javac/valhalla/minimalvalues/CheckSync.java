/*
 * @test /nodynamiccopyright/
 * @summary May not synchronize on value types
 * @modules jdk.incubator.mvt
 * @compile/fail/ref=CheckSync.out --should-stop:ifError=PARSE -XDrawDiagnostics -Werror -Xlint:values  CheckSync.java
 */

/* Note: ATM, value types do not have jlO in their lineage. So they anyway
   cannot synchronize using the methods declared on jlO.
*/
@jdk.incubator.mvt.ValueCapableClass
public final class CheckSync {
    @jdk.incubator.mvt.ValueCapableClass
    final class Val {

        void foo() {
            // All calls below are bad.
            wait();
            wait(10);
            wait(10, 10);
            notify();
            notifyAll();
            finalize();
            clone();
        }
    }

    final Val val = new Val();

    void test() throws InterruptedException {
        // All calls below are bad.
        val.wait();
        val.wait(10);
        val.wait(new Integer(10));
        val.wait(new Long(10));
        val.wait(10L);
        val.wait(10L, 10);
        val.wait("Hello");
        val.notify();
        val.notifyAll();
    }
}
