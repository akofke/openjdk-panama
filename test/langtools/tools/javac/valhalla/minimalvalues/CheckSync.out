- compiler.warn.incubating.modules: jdk.incubator.mvt
- compiler.err.warnings.and.werror
CheckSync.java:38:12: compiler.err.cant.apply.symbols: kindname.method, wait, java.lang.String,{(compiler.misc.inapplicable.method: kindname.method, java.lang.Object, wait(), (compiler.misc.arg.length.mismatch)),(compiler.misc.inapplicable.method: kindname.method, java.lang.Object, wait(long), (compiler.misc.no.conforming.assignment.exists: (compiler.misc.inconvertible.types: java.lang.String, long))),(compiler.misc.inapplicable.method: kindname.method, java.lang.Object, wait(long,int), (compiler.misc.arg.length.mismatch))}
CheckSync.java:18:17: compiler.warn.value.does.not.support: calling java.lang.Object's method: wait
CheckSync.java:19:17: compiler.warn.value.does.not.support: calling java.lang.Object's method: wait
CheckSync.java:20:17: compiler.warn.value.does.not.support: calling java.lang.Object's method: wait
CheckSync.java:21:19: compiler.warn.value.does.not.support: calling java.lang.Object's method: notify
CheckSync.java:22:22: compiler.warn.value.does.not.support: calling java.lang.Object's method: notifyAll
CheckSync.java:23:21: compiler.warn.value.does.not.support: calling java.lang.Object's method: finalize
CheckSync.java:24:18: compiler.warn.value.does.not.support: calling java.lang.Object's method: clone
CheckSync.java:32:17: compiler.warn.value.does.not.support: calling java.lang.Object's method: wait
CheckSync.java:33:17: compiler.warn.value.does.not.support: calling java.lang.Object's method: wait
CheckSync.java:34:17: compiler.warn.value.does.not.support: calling java.lang.Object's method: wait
CheckSync.java:35:17: compiler.warn.value.does.not.support: calling java.lang.Object's method: wait
CheckSync.java:36:17: compiler.warn.value.does.not.support: calling java.lang.Object's method: wait
CheckSync.java:37:17: compiler.warn.value.does.not.support: calling java.lang.Object's method: wait
CheckSync.java:39:19: compiler.warn.value.does.not.support: calling java.lang.Object's method: notify
CheckSync.java:40:22: compiler.warn.value.does.not.support: calling java.lang.Object's method: notifyAll
- compiler.note.deprecated.filename: CheckSync.java
- compiler.note.deprecated.recompile
2 errors
16 warnings
